base/linalg/cholesky.jl

```diff
@@ -462,7 +467,7 @@ function A_ldiv_B!(C::CholeskyPivoted, B::StridedMatrix)
 end
 
 function det(C::Cholesky)
-    C.info == 0 || throw(PosDefException(C.info))
+    issuccess(C) || throw(PosDefException(C.info))
```

Comment:
perhaps clearer to check `isposdef` since we throw an `PosDefException` here? (eg. #22245 )? same below